### PR TITLE
New Lesson - Automating with JET

### PR DESCRIPTION
**Pull request by jnpr-raylam**
It's ready for review. Thanks.

Stage1: JET rpc/notification configuration
Stage2: Receive events from notification service using MQTT client
Stage3: IDL compilation and basic GRPC JET testing
Stage4: JET Firewall API to create/update firewall filter
Stage5: Closed loop automation with JET

Closes https://github.com/nre-learning/antidote/issues/164

### CODE REVIEW

**Comment by Mierdin** [comment]
Hey @jnpr-raylam thanks for this. I had hoped to be able to review this in time for the upcoming release this weekend, but there were a few problem that came up that needed addressed. Once v0.3.0 comes out in the next few days, I'll turn around and get your lesson reviewed so we can put it into v0.3.1, which will ideally drop the week after. Thanks for your patience.

**Comment by jnpr-raylam** [comment]
@Mierdin any update on this? Thanks!

**Comment by Mierdin** [comment]
Thanks @jnpr-raylam. Will begin a full review shortly. Thanks for working on this.

Can you reword the title, though? "Automating with JET" doesn't really describe what's going on here, or what openconfig really is. "automating" can apply broadly to anything on this site. Maybe something like "The Junos Extensibility Toolkit (JET)"?

File: lessons/lesson-25/stage1/guide.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+* Notification - An application can receive asynchronous notification of events happening on Junos OS. (publish-subscribe model. MQTT based)
+
+For more informations about the JET can be found <a href="https://www.juniper.net/documentation/en_US/jet18.4/topics/concept/jet-architecture.html" target="_blank">here</a>.
+
+In this lab we are going to explore a off-box python JET application.
+
+#### Config the Junos device for JET
+First of all, to run an off-box JET application, we need to enable the request-response configuration on the Junos OS device.
+
+The gRPC can be run in clear-text mode _(insecure! that's why it is hidden and for lab test only!)_ or SSL encrypted mode for enhanced security. For simpilicity we'll go with clear-text in the lab. More information about gRPC over SSL can be found <a href="https://www.juniper.net/documentation/en_US/jet18.4/topics/topic-map/jet-off-box-apps.html" target="_blank">here</a>.
```
Comment (Mierdin) [inline lessons/lesson-25/stage1/guide.md]:
```suggestion
The gRPC can be run in clear-text mode _(insecure! that's why it is hidden and for lab test only!)_ or SSL encrypted mode for enhanced security. For simplicity we'll go with clear-text in the lab. More information about gRPC over SSL can be found <a href="https://www.juniper.net/documentation/en_US/jet18.4/topics/topic-map/jet-off-box-apps.html" target="_blank">here</a>.
```

File: lessons/lesson-25/stage1/guide.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
+## Automating with JET
```
Comment (Mierdin) [inline lessons/lesson-25/stage1/guide.md]:
```suggestion
## Junos Extensibility Toolkit (JET)
```

File: lessons/lesson-25/stage2/guide.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
+## Automating with JET
```
Comment (Mierdin) [inline lessons/lesson-25/stage2/guide.md]:
```suggestion
## Junos Extensibility Toolkit (JET)
```

File: lessons/lesson-25/stage3/guide.md [hunk 4]
```diff
@@ -0,0 +1,125 @@
+## Automating with JET
```
Comment (Mierdin) [inline lessons/lesson-25/stage3/guide.md]:
```suggestion
## Junos Extensibility Toolkit (JET)
```

File: lessons/lesson-25/stage4/guide.md [hunk 5]
```diff
@@ -0,0 +1,132 @@
+## Automating with JET
```
Comment (Mierdin) [inline lessons/lesson-25/stage4/guide.md]:
```suggestion
## Junos Extensibility Toolkit (JET)
```

File: lessons/lesson-25/stage5/guide.md [hunk 6]
```diff
@@ -0,0 +1,96 @@
+## Automating with JET
```
Comment (Mierdin) [inline lessons/lesson-25/stage5/guide.md]:
```suggestion
## Junos Extensibility Toolkit (JET)
```

File: lessons/lesson-25/syringe.yaml [hunk 7]
```diff
@@ -0,0 +1,39 @@
+---
+lessonName: Automating with Junos Extension Toolkit (JET)
```
Comment (Mierdin) [inline lessons/lesson-25/syringe.yaml]:
```suggestion
lessonName: Junos Extensibility Toolkit (JET)
```

File: lessons/lesson-25/syringe.yaml [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+lessonId: 25
+category: configuration
+tier: local
+juniperSpecific: true
```
Comment (Mierdin) [inline lessons/lesson-25/syringe.yaml]:
```suggestion
```

not needed anymore

**Comment by Mierdin** [comment]
A quick update - I'm having a discussion in https://github.com/nre-learning/syringe/pull/93 regarding one of things that's standing in the way of us using the new full vqfx image. Once that conversation is worked out I believe we'll be ready to use the new image. Stay tuned there for updates for now.

In the meantime, I'd like to keep this PR open so we can merge it when it's totally ready to go. I'd feel comfortable merging it now, but I'd have to keep it in PTR until the new image is ready to use, at which point we'd have to update this lesson to use it anyways. I'm still committed to getting this in the next release so we're really all just waiting for this image stuff to get hashed out and then all the pieces will fall into place. Thanks for your patience.

**Comment by Mierdin** [comment]
I've decided to merge this now, as the lesson itself is solid, and the image work being done in #208 and #206 can happen in parallel. I'd like to get this lesson merged sooner rather than later, because I am pretty sure the only PR that's waiting for this image work is #191, and it doesn't make sense to hold up the work done in this PR as well as #184 just for that.

Thanks again for this work!

**Pull request merged**